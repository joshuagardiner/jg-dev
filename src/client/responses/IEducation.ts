/**
 * The response object the API returns.
 */
export interface IEducationResponse {
  /**
   * The course studied.
   */
  course?: string;

  /**
   * The faculty studied at.
   */
  faculty: string;

  /**
   * The grades achieved.
   */
  grades?: IGrade[];

  /**
   * The identifier.
   */
  id: string;

  /**
   * The tenure of study.
   */
  tenure: string;
}

/**
 * The response object the API returns.
 */
interface IGrade {
  /**
   * The identifier.
   */
  id: string;

  /**
   * The value.
   */
  value: string;
}
