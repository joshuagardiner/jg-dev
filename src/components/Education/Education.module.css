.container {
  margin-bottom: 8.5%;
}
