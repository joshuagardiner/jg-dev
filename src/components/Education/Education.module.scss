.course {
  font-size: 1.125rem;
  text-transform: uppercase;
}

.divider {
  color: #3c4858;
  width: 5%;
}

.faculty {
  font-size: 1rem;
  text-transform: uppercase;
}

.grade {
  font-size: 0.75rem;
}

.tenure {
  font-size: 0.75rem;
  text-transform: uppercase;
}

@media screen and (width <= 425px) {
  .course {
    font-size: 0.75rem;
  }

  .faculty {
    font-size: 0.66rem;
  }

  .grade {
    font-size: 0.5rem;
    line-height: 1rem;
  }

  .tenure {
    font-size: 0.5rem;
  }
}
