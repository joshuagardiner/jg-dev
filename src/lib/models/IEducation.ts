/**
 * The Education model.
 */
export interface IEducation {
  /**
   * The course studied.
   */
  courseOfStudy: string;

  /**
   * The grades achieved.
   */
  grades: string[];

  /**
   * The place of study.
   */
  placeOfStudy: string;

  /**
   * The tenure of study.
   */
  tenure: string;
}
