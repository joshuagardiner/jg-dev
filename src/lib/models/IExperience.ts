/**
 * The Experience model.
 */
export interface IExperience {
  /**
   * The company.
   */
  company: string;

  /**
   * The content.
   */
  content: IExperienceContent[];

  /**
   * The identifier.
   */
  id: string;

  /**
   * The content's priority.
   */
  priority: boolean;

  /**
   * The role.
   */
  role: string;

  /**
   * The tenure.
   */
  tenure: string;
}

/**
 * The ExperienceContent model.
 */
export interface IExperienceContent {
  /**
   * The identifier.
   */
  id: string;

  /**
   * The value.
   */
  value: string;
}
