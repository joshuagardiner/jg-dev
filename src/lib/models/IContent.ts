import { IExperience } from "./IExperience";
import { ISkills } from "./ISkills";
import { ISummary } from "./ISummary";

/**
 * The Content model.
 */
export interface IContent {
  /**
   * The experiences content.
   */
  experiences: IExperience[];

  /**
   * The skills content.
   */
  skills: ISkills;

  /**
   * The summary content.
   */
  summary: ISummary;
}
