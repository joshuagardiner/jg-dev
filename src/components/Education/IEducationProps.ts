import { IGrade } from "../../lib/models/IEducation";

/**
 * The properties available to the Education component.
 */
export interface IEducationProps {
  /**
   * The course studied.
   */
  course: string;

  /**
   * The location of study.
   */
  faculty: string;

  /**
   * The grades achieved.
   */
  grades: IGrade[];

  /**
   * The timeframe of study.
   */
  tenure: string;
}
