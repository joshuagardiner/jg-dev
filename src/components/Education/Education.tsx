import { IGrade } from "../../lib/models/IEducation";
import { IEducationProps } from "./IEducationProps";
import styles from "./Education.module.scss";

/**
 * The Education component is responsible for rendering a single academic
 * detail within the application body.
 *
 * @param props the properties that can be passed into the component.
 * @returns the Education component.
 */
export const Education: React.FC<IEducationProps> = (props: IEducationProps) => {
  const { course, faculty, grades, tenure } = props;

  return (
    <div data-testid={"education"}>
      <h3 className={styles.course} data-testid={"education-course"}>
        {course}
      </h3>
      <h4 className={styles.faculty} data-testid={"education-faculty"}>
        {faculty}
      </h4>
      <h5 className={styles.tenure} data-testid={"education-tenure"}>
        {tenure}
      </h5>
      <hr className={styles.divider} />
      {grades.map((grade: IGrade) => (
        <p key={grade.id} className={styles.grade} data-testid={"education-grade"}>
          {grade.value}
        </p>
      ))}
    </div>
  );
};
