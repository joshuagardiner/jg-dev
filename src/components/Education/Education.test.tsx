import { render, screen } from "@testing-library/react";
import { IEducation } from "../../lib/models/IEducation";
import { Education } from "./Education";

describe("Education", () => {
  it("renders the Education component correctly.", () => {
    const education: IEducation[] = [
      {
        courseOfStudy: "University",
        grades: [],
        placeOfStudy: "Bedroom under the stairs",
        tenure: "6 weeks"
      },
      {
        courseOfStudy: "School",
        grades: ["A's", "B's", "C's"],
        placeOfStudy: "Classroom",
        tenure: "5 years"
      }
    ];
    render(<Education education={education} />);

    const educationComponent = screen.getByTestId("education-component");
    const educationTitle = screen.getByTestId("education-title");
    const educationCourse = screen.getByTestId("education-course");
    const educationFaculties = screen.getAllByTestId("education-faculty");
    const educationTenures = screen.getAllByTestId("education-tenure");
    const educationGrades = screen.getByTestId("education-grades");

    expect(educationComponent).toBeDefined();
    expect(educationTitle.textContent).toEqual("Education");
    expect(educationCourse.textContent).toEqual("University");
    expect(educationFaculties).toHaveLength(2);
    expect(educationTenures).toHaveLength(2);
    expect(educationFaculties[0].textContent).toEqual("Bedroom under the stairs");
    expect(educationTenures[0].textContent).toEqual("6 weeks");
    expect(educationFaculties[1].textContent).toEqual("Classroom");
    expect(educationTenures[1].textContent).toEqual("5 years");
    expect(educationGrades.textContent).toEqual("A'sB'sC's");
  });
});
