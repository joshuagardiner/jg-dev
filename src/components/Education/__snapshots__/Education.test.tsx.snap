// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education renders the Education component unchanged. 1`] = `
<div>
  <div
    class="container"
    data-testid="education-component"
  >
    <h2
      class="title"
      data-testid="education-title"
    >
      Education
    </h2>
    <div
      class="tcl-card"
      data-testid="card"
      style="height: auto; margin-top: auto; max-width: 100%;"
    >
      <div
        class="tcl-card__content"
        data-testid="card-content"
      >
        <h3
          class="course"
          data-testid="education-course"
        >
          BSc (Hons) COMPUTING
        </h3>
        <h4
          class="faculty"
          data-testid="education-faculty"
        >
          Manchester Metropolitan University
        </h4>
        <h5
          class="tenure"
          data-testid="education-tenure"
        >
          September '11 - June '16
        </h5>
        <div
          class="tcl-card"
          data-testid="card"
          style="height: auto; margin-top: 1vw; max-width: 100%;"
        >
          <div
            class="tcl-card__content"
            data-testid="card-content"
          >
            <h4
              class="faculty"
              data-testid="education-faculty"
            >
              The Brooksbank School and Sixth Form
            </h4>
            <h5
              class="tenure"
              data-testid="education-tenure"
            >
              September '03 - June '10
            </h5>
            <p
              class="grades"
              data-testid="education-grades"
            >
              A-level's: Information technology, Geography
              <br />
              AS-level's: Business studies, Physical education
              <br />
              GCSE's: 10 [A-C]
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
